### PR TITLE
sock: expose configuration file

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

Adapts #11558  in accordance to #10566.

When reviewing the PR I didn't take care in enforcing #10566. I'm pushing this PR to fix that overlook on my side.

### Testing procedure

Run CI

### Issues/PRs references

#11558

### CODE REVIEW

File: sys/include/net/sock/config.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2018 Freie Universität Berlin
+ *               Copyright (C) 2019 Inria
```
Comment (kaspar030) [inline sys/include/net/sock/config.h]: this line is off

Comment (fjmolinas) [reply sys/include/net/sock/config.h]: solved

**Comment by kaspar030** [comment]
@miri64 does this have to be within its own config.h?

Isn't the important change to prefix the defines with ```CONFIG_```?

**Comment by miri64** [comment]
AFAIK it doesn't has to be. The important thing is that it is within a doxygen group (which doesn't has to be in its own file). IIRC for gcoap we did it like that.

> Isn't the important change to prefix the defines with CONFIG_?

I'm not sure this is anymore this is a sensible, scalable change. Just take `pkg` for an example: Technically they contain compile-configuration for RIOT. Exposing them all as `CONFIG_…` provides a huge maintenance burden for us. So why not find a more unified way to expose/document/configure them? Let's discuss this further in #10566, however.

**Comment by fjmolinas** [comment]
@miri64 So in this case only the doxygen config group is enough?

**Comment by miri64** [comment]
> @miri64 So in this case only the doxygen config group is enough?

Yepp

**Comment by fjmolinas** [comment]
@miri64 @kaspar030 changes accordingly and all is green.

**Comment by fjmolinas** [comment]
@miri64 thanks for the review!

**Pull request merged**